### ISSUE TITLE
App notifications standardization

### ISSUE THREAD

**Issue by zamentur** [YunoHost/issues#1957]
This is completely a draft:

Today we discussed about app packaing v2 with Aleks, here you can find an attempt to standardize the messages we want the admin to read.

We can imagine a tree like this in our app:
```
doc
├── ADMIN.md
├── notifications
│   ├── disclaimer.d
│   │   ├── non_free_addons.md
│   │   └── artificial_limitation.md
│   ├── post_install.d
│   │   ├── 01.access_admin.md
│   │   └── 02.unautomated_install_steps.md
│   ├── post_upgrade.d
│   │   ├── 1.0.3.md
│   │   └── any.md
│   └── pre_upgrade.d
│       ├── 1.0.3.md
│       └── any.md
└── screenshots
```

## Disclaimer messages types (antifeatures)

Here you can find a list (may be not exhaustive of normalized antifeatures):
   - tracking.md: tracks and/or reports users activity to somewhere, even when it can be turned off 
   - non_free_upstream.md: upstream source code is not libre / not totally free
   - non_free_addons.md: promotes other non-libre apps or plugins
   - non_free_dependencies.md: needs a non-libre app to work
   - non_free_network_services.md:  encourage the use of private platform
   - known_vulnerabilities.md: some important vuln unfixed
   - artificial_limitation.md: the app contains some artificial limitations (mobile support disable in free version...)
   - limitation.md: important limitation to know when using the app
   - alpha_software.md: the app is in an early stage of development (alpha/beta)
   - deprecated_software.md: not maintained at all, totally end of life, replaced by an other apps...
   - ads.md: some ads are in the app
   - bad_reputation.md: known as a bigger risk for security or some modules could be dangereous or problematic for privacy, unreliable etc...

ALl thos messages will be displayed like a warning alert on top of the unexistent app description page (after selecting the app in catalog).

About rtfm messages and others things to print to the user during installation process, we suggest to use the install question of the manifest (or sso/ldap/ressources properties) instead to use disclaimer.

## Post-installation messages types
Here an exhaustive list of what can contains the postinstall message:
- link to the admin and to the app...
- DNS config
- Port forwarding
- create first admin user
- how to use the app if it's not obvious
- credential transmission
- post-post installation crazy (db init...)
- other
 
Post-installation messages could be displayed as a popup at the end of the installation of an app

## Pre-upgrade messages types
- disable some apps plugin / or warn about unsupported apps plugin
- wordpress use case (upstream version won't be updated)
- operation to do manually (like with mailman one time)
- change permission behavior
- depreciation (upstream app id recommended change (owncloud -> nextcloud)

## Post-upgrade messages types

- reactive some plugins
- check wordpress version
- warn about a surprising behavior / major change in the app config

**Comment by yalh76** [comment]
Why not generalising the notifications, having something like

```
doc
├── ADMIN.md
├── notifications
│   ├── install_pre.d
│   │   ├── non_free_addons.md
│   │   └── artificial_limitation.md
│   ├── install_post.d
│   │   ├── 01.access_admin.md
│   │   └── 02.unautomated_install_steps.md
│   ├── upgrade_post.d
│   │   ├── 1.0.3.md
│   │   └── any.md
│   └── upgrade_pre.d
│       ├── 1.0.3.md
│       └── any.md
└── screenshots
```
 

And having auto-readme retrieving the content of install_pre.d

**Comment by alexAubin** [comment]
Closing because it's basically implemented in 11.1, though webadmin integration is still wip
